### ISSUE TITLE
Pagination missing aria-current label

### ISSUE THREAD

**Issue by shainanigans** [StackExchange/Stacks#1054]
**Describe the bug**
In components like pagination, an [`aria-current` attribute](https://www.w3.org/TR/wai-aria-1.1/#aria-current) is required to indicate to screen reader users which page is currently active.

**To Reproduce**
1. Go to the [Pagination](https://stackoverflow.design/product/components/pagination/) example
2. Use a screen reader of your choosing to listen to the announcement of the example component
3. Note that the currently selected page is not announced as such

**Expected behavior**
The currently selected page (which gets the `is-selected` CSS class) should be announced.
